### PR TITLE
refactor: generate function refactored to reduce cognitive complexity

### LINKED ISSUE #1040: Fix cognitive complexity issue
So `generate` command from https://github.com/asyncapi/generator/blob/master/lib/generator.js#L163 grew a bit and our code [smell checks complain](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=1039&id=asyncapi_generator&open=AYrVno9ob6ubVuTTBoqq)

I think the point is valid, maybe for maintainers it is not a big problem, but for contributors, for sure.

We need help from someone to figure out how best refactor that part

### PR DESCRIPTION
**Pull request by swastiksuvam55**
`generate` function refactored to smaller individual functions to reduce cognitive complexity

Related issue(s):
Fixes #1040

### CODE REVIEW

File: lib/generator.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ class Generator {
    * @return {Promise}
    */
   // eslint-disable-next-line sonarjs/cognitive-complexity
-  async generate(asyncapiDocument, parseOptions = {}) {
+  /* async generate(asyncapiDocument, parseOptions = {}) {
```
Comment (jonaslagoni) [inline lib/generator.js:163]: Please remove any commented out code 🙂

File: lib/generator.js [hunk 2]
```diff
@@ -207,6 +207,108 @@ class Generator {
       await this.generateDirectoryStructure(this.asyncapi);
       await this.launchHook('generate:after');
     }
+  } */
+
+  // Main method for generating code based on AsyncAPI document
```
Comment (jonaslagoni) [inline lib/generator.js:212]: Please switch to the jsdoc comments (for all functions) and use the same as the old one for the function 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Just a few comments 💪

**Comment by jonaslagoni** [comment]
You also have a linting issue 🙂

**Comment by swastiksuvam55** [comment]
@jonaslagoni  Thank you for the much-needed feedback, I've made the changes can you please review it again?

File: lib/generator.js [hunk 3]
```diff
@@ -136,67 +136,174 @@ class Generator {
     });
   }
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
```
Comment (jonaslagoni) [inline lib/generator.js:139]: Please remove this 🙂

File: lib/generator.js [hunk 4]
```diff
@@ -136,67 +136,174 @@ class Generator {
...
+   * console.log('Done!');
+   *
+   * @param {AsyncAPIDocument | string} asyncapiDocument - AsyncAPIDocument object to use as source.
+   * @param {Object} [parseOptions={}] - AsyncAPI Parser parse options.
+   *   Check out {@link https://www.github.com/asyncapi/parser-js|@asyncapi/parser} for more information.
+   *   Remember to use the right options for the right parser depending on the template you are using.
+   * @return {Promise<void>} A Promise that resolves when the generation is completed.
    */
-  // eslint-disable-next-line sonarjs/cognitive-complexity
+
```
Comment (jonaslagoni) [inline lib/generator.js:155]: No new lines between the jsdoc and function 🙂

File: lib/generator.js [hunk 5]
```diff
@@ -136,67 +136,174 @@ class Generator {
...
-   *   })
-   *   .catch(console.error);
-   *
-   * @example <caption>Using async/await</caption>
-   * try {
-   *   await generator.generate(myAsyncAPIdocument);
-   *   console.log('Done!');
-   * } catch (e) {
-   *   console.error(e);
-   * }
```
Comment (jonaslagoni) [inline lib/generator.js:156]: You can leave the different examples, no need to remove it 🙂

File: lib/generator.js [hunk 6]
```diff
@@ -136,67 +136,174 @@ class Generator {
...
     const isParsableCompatible = asyncapiDocument && typeof asyncapiDocument === 'string';
+
     if (!isAlreadyParsedDocument && !isParsableCompatible) {
       throw new Error('Parameter "asyncapiDocument" must be a non-empty string or an already parsed AsyncAPI document.');
     }
+
     this.asyncapi = this.originalAsyncAPI = asyncapiDocument;
+  }
 
+  // Setup the output based on the configured options
```
Comment (jonaslagoni) [inline lib/generator.js:185]: No need for this.

File: lib/generator.js [hunk 7]
```diff
@@ -136,67 +136,174 @@ class Generator {
...
+     * Installs and sets up the template for code generation.
+     *
+     * This function installs the specified template using the provided installation option,
+     * sets up the necessary directory paths, loads the template configuration, and returns
+     * information about the installed template.
+     *
+     * @async
+     * @returns {Promise<{ templatePkgName: string, templatePkgPath: string }>}
+     *   A promise that resolves to an object containing the name and path of the installed template.
+  */
```
Comment (jonaslagoni) [inline lib/generator.js:245]: Incorrect indentation 🙂

File: lib/generator.js [hunk 8]
```diff
@@ -205,10 +312,21 @@ class Generator {
...
   }
 
+  /**
+   * Executes the 'generate:after' hook.
+   *
+   * Launches the after-hook to perform additional actions after code generation.
+   *
+   * @async
+   * @returns {Promise<void>} A promise that resolves when the after-hook execution is completed.
+  */
```
Comment (jonaslagoni) [inline lib/generator.js:325]: Incorrect indentation

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Few more changes is needed 🙂

**Comment by jonaslagoni** [comment]
@all-contributors please add @swastiksuvam55 for code

**Pull request merged**